src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+   * Your package's GitHub _repository_ from which you publish the
+     package.
+   * A _`tag-pattern`_ that git tags must match to permit publishing.
+1. Create a GitHub Action workflow for publishing in GitHub.
+1. Create a `pubspec.yaml` file in your repository.
+   This file requires a `version` key with a value that matches the
+   `tag-pattern`.
+1. Create a git tag for your latest commit.
+   This tag also must match the `tag-pattern`.
+
+{{site.alert.note}}
+  GitHub Actions can publish to `pub.dev` only when a push of a git tag
+  to GitHub triggers the Action _workflow_.
+  If a GitHub Actions triggers publishing without a tag, `pub.dev`
+  rejects it.
+  This ensures that events that don't trigger publish result in
+  publishing new versions.
+{{site.alert.end}}
+
+
+## Configure automated publishing from GitHub Actions on pub.dev
+
+### Check for required `pub.dev` permissions
+
+To automate publication from GitHub Actions to `pub.dev`, you need one
+of the following roles:
+
+* An _uploader_ on the package
+* An _admin_ of the publisher if a publisher owns the package
+
+### Configure automated publishing
+
+To enable `pub.dev` to accept GitHub Action publishing, complete the
+following steps.
+
+1. Navigate to the **Admin** tab (`pub.dev/packages/<package>/admin`).
+1. Find the **Automated publishing** section.
+
+   ![pub.dev Publishing from GitHub Actions form](pub-dev-gh-setup.png)
+
+1. Click **Enable publishing from GitHub Actions**.
+   Additional fields display when you select this checkbox.
+1. Write the GitHub repository in the **Repository** field.
+
+   A _repository_ contains all of your package's files on GitHub.
+   This field expects a format of `<organization>/<repository>`.
+   For example: If the URL for your repository resembles
+   `https://github.com/dart-lang/pana`, write `dart-lang/pana` in the
+   **Repository** field.
+
+1. Write the _`tag-pattern`_ into the **Tag pattern** field.
+
+   A `tag-pattern` represents the version of the package.
+   This field expects a string that includes the literal
+   `{% raw %}{{version}}{% endraw %}`.
+   Github Actions can publish your package only if a push of a git tag
+   matching this `tag-pattern` triggers the Action.
+   The package's `pubspec.yaml` file requires the `version` key.
+   This key's value must match the `{% raw %}{{version}}{% endraw %}`
+   in the `tag-pattern`.
+
+1. Click **Update**.
+
+<!-- **Example:** You wrote your `tag-pattern` as `v{{version}}`.
+You then tag this commit and push it.
+
+```shell
+git tag v1.2.3 && git push v1.2.3
+```
+
+This push triggers Github Actions to publish version `1.2.3` of
+your package. -->
+
+If your repository contains multiple packages, give each a separate
+`tag-pattern`. Consider using a `tag-pattern` like
+`my_package_name-v``{% raw %}{{version}}{% endraw %}` for a package
+named `my_package_name`.
+
+## Prepare your package repository for automated publishing
```

Comment:
This is confusing.... you're preparing the `pubspec.yaml` for publishing a specific version, not for configuring automated publishing.